icd/api/include/pipeline_binary_cache.h

```diff
@@ -152,7 +176,32 @@ class PipelineBinaryCache
 
     void FreePipelineBinary(const void* pPipelineBinary);
 
-    void Destroy() { this->~PipelineBinaryCache(); }
+    void* AllocMem(size_t memSize,
+                   size_t memAlignment = VK_DEFAULT_MEM_ALIGN,
+                   VkSystemAllocationScope allocationScope = VK_SYSTEM_ALLOCATION_SCOPE_OBJECT) const
+    {
+        VK_ASSERT(memSize > 0);
+        return m_pAllocationCallbacks->pfnAllocation(m_pAllocationCallbacks->pUserData,
+                                                     memSize,
+                                                     memAlignment,
+                                                     allocationScope);
+    }
+
+    void FreeMem(void* pMem) const
+    {
+        if (pMem != nullptr)
+        {
+            m_pAllocationCallbacks->pfnFree(m_pAllocationCallbacks->pUserData, pMem);
+        }
+    }
+
+    void DestroyAndFree()
+    {
+        VkAllocationCallbacks* pAllocationCallbacks = m_pAllocationCallbacks;
+        void* pMem = this;
+        this->~PipelineBinaryCache();
+        pAllocationCallbacks->pfnFree(pAllocationCallbacks->pUserData, pMem);
```

Comment:
I think you should just move the free of the memory into the existing Destroy().  That's how we typically do it anyhow (ex: ImageView::Destroy()).  And keep Destroy in the cpp.